lib/sablon/image.rb

```diff
@@ -0,0 +1,17 @@
+module Sablon
+  class Image
+    include Singleton
+    attr_reader :definitions
+
+    Definition = Struct.new(:name, :data, :rid) do
+      def inspect
+        "#<Image #{name}:#{data}:#{rid}"
+      end
+    end
+
+    def self.create_by_path(path, random = nil)
+      image_name = "#{random || Random.new_seed}-#{Pathname.new(path).basename.to_s}"
```

Comment:
I have had some issues with the encoding of filenames with special chars in the archive on windows, so I would do a `.gsub(/[^0-9A-Za-z_\.\-]/, '')` on the basename or something similar. This fixed a broken document for me. Is there a reason to have the original basename included in `image_name` in the first place?